Sources/Internal/VisibleItemsProvider.swift

```diff
@@ -242,7 +248,7 @@ final class VisibleItemsProvider {
 
     // Handle pinned day-of-week layout items
     if case .vertical(let options) = content.monthsLayout, options.pinDaysOfWeekToTop {
-      handlePinnedDaysOfWeekIfNeeded(yContentOffset: bounds.minY, context: &context)
+      handlePinnedDaysOfWeekIfNeeded(yContentOffset: offset.y, context: &context)
```

Comment:
we don't want to use the extended bounds for this, otherwise we'll render these pinned items off screen